samples/graph-activity-feed-broadcast/nodejs/server.js

```diff
@@ -0,0 +1,256 @@
+              "source": "text",
+              "value": req.body.title,
+              "webUrl": 'https://teams.microsoft.com/l/entity/' + appId + '/broadcast?context=' + encodedContext
+            },
+            "activityType": "approvalRequired",
+            "previewText": {
+              "content": "Broadcast by"+ req.body.userName
+            },
+            "templateParameters": [
+              {
+                "name": "approvalTaskId",
+                "value": req.body.title
+              }
+            ]
+          };
+        if (userAppId == undefined) {
+          const broadcastAppId = {
+            "teamsApp@odata.bind": "https://graph.microsoft.com/v1.0/appCatalogs/teamsApps/" + appId
+          };
+
+          axios.post("https://graph.microsoft.com/v1.0/users/" + userList.data.value[i].id + "/teamwork/installedApps", broadcastAppId, {
+            headers: {
+              "accept": "application/json",
+              "contentType": 'application/json',
+              "authorization": "bearer " + delegatedToken
+            }
+          });
+
+          axios.post("https://graph.microsoft.com/v1.0/users/" + userList.data.value[i].id + "/teamwork/sendActivityNotification", postData, {
+            headers: {
+              "accept": "application/json",
+              "contentType": 'application/json',
+              "authorization": "bearer " + applicationToken
+            }
+          })
+          .then(res => {
+            console.log(`statusCode: ${res.status}`)
+            if(res.status == 429)
+            {
+              polly()
+              .handle(function(err) {
+                  return err.statusCode === 429;
+              })
+              .waitAndRetry(3)
+              .executeForNode(function () {
+                axios.post("https://graph.microsoft.com/v1.0/users/" + userList.data.value[i].id + "/teamwork/sendActivityNotification", postData, {
+                  headers: {
+                    "accept": "application/json",
+                    "contentType": 'application/json',
+                    "authorization": "bearer " + applicationToken
+                  }
+                })
+              }, function (err, data) {
+                  if (err) {
+                      console.error('Failed trying twice with a 100ms delay', err)
+                  } else {
+                      console.log(data)
+                  }
+              });
+            }
+          })
+        }
+        else {
+          axios.post("https://graph.microsoft.com/v1.0/users/" + userList.data.value[i].id + "/teamwork/sendActivityNotification", postData, {
+            headers: {
+              "accept": "application/json",
+              "contentType": 'application/json',
+              "authorization": "bearer " + applicationToken
+            }
+          })
+          .then(res => {
+            console.log(`statusCode: ${res.status}`)
+            if(res.status == 429)
+            {
+              polly()
```

Comment:
Shall we move this code to common method?

Reply:
moved to a common method